snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -284,18 +277,23 @@ def get_dirty_report(self, step):
             options = state.diff_project_options_of_interest(
                 self._project_options)
 
-            # If dependencies have changed since this step ran, then it's dirty
-            # and needs to run again.
-            dependencies = state.changed_dependencies
-
-            if properties or options or dependencies:
-                return DirtyReport(properties, options, dependencies)
+            if properties or options:
+                return DirtyReport(
+                    dirty_properties=properties, dirty_project_options=options)
 
         return None
 
     def should_step_run(self, step, force=False):
         return force or self.is_clean(step)
 
+    def step_timestamp(self, step):
+        try:
+            return os.stat(
```

Comment:
I can tell you like stairs :taunt: (emoji request :smile: )

Reply:
Yikes, fixed that a bit.